modin/experimental/core/execution/native/implementations/hdk_on_native/test/utils.py

```diff
@@ -161,13 +161,17 @@ def __init__(self, *dfs):
             df.shape  # to trigger real execution
             if df.empty:
                 continue
-            partition = df._query_compiler._modin_frame._partitions[0][0]
+            modin_frame = df._query_compiler._modin_frame
+            partition = modin_frame._partitions[0][0]
             if partition.frame_id is not None:
                 continue
             frame = partition.get()
             if isinstance(frame, (pandas.DataFrame, pandas.Series)):
-                frame_id = DbWorker().import_pandas_dataframe(frame)
-            elif isinstance(frame, pa.Table):
+                frame = pa.Table.from_pandas(frame)
+            if isinstance(frame, pa.Table):
+                _, cols = modin_frame._partition_mgr_cls._get_unsupported_cols(frame)
+                if len(cols) != 0:
```

Comment:
What happens with the frames that have unsupported cols? Do we just lose them?

Reply:
The frames are not imported to HDK and should be default to pandas.

Reply:
Such frames are just not imported to HDK because their import might cause an error and is useless anyway since we are supposed to fall back to Pandas in case of unsupported columns.

The frame actually should have a flag that it has such columns, so probably it would be enough to check it instead. I initially supposed that we might need this unsupported column check for cases when there is a Pandas dataframe in a partition. But now I recall that we should have Pandas dataframe in a partition only when there are unsupported columns in it, so probably there is no reason to convert to Arrow here.

Reply:
Also, I wonder if no one of the passed dfs has been imported into HDK, then we might have an issue here.
https://github.com/modin-project/modin/blob/ff53445e1baa916d4b4ac6f2ecab21b86f5ccf02/modin/experimental/core/execution/native/implementations/hdk_on_native/test/test_dataframe.py#L2067
https://github.com/modin-project/modin/blob/ff53445e1baa916d4b4ac6f2ecab21b86f5ccf02/modin/test/interchange/dataframe_protocol/hdk/utils.py#L76